### PR TITLE
Document strategy for branching, merging, and commit messages

### LINKED ISSUE #30: Pick branch strategy
I guess as most projects _JUnit Pioneer_ will favor developing on branches. Should this be made mandatory even for seemingly small fixes?

How should branches be merged? Should regular merges, resulting in merge commits, be favored or rebases? Related to that should commits be cleaned up or even squashed or is a more chaotic commit history ok?

- [x] decide on how to branch and merge
- [x] document in [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)

### LINKED ISSUE #31: Which commits should reference issues?
When starting out to work on the code base I made every commit reference the issue it belonged to. That has the great advantage that it is very easy to jump from Git history to pertaining issues. It has the disadvantage that it spams the issue with links to the commit history. In a particularly unnerving move, these references get _duplicated_ once the PR gets merged and all commits appear on master.

An alternative would be to only mention issues in the merge commits.

- [ ] decide on where to include issue references
- [ ] document in [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by nicolaiparlog**
After coming to terms on [how to branch and merge](https://github.com/junit-pioneer/junit-pioneer/issues/30#issuecomment-298891508) (closes #30) and [when to mention issues](https://github.com/junit-pioneer/junit-pioneer/issues/31#issuecomment-298892406) (closes #31), I am documenting both decisions in `CONTRIBUTING.md`.

---

I hereby agree to the terms of the [Io Contributor License Agreement](../CONTRIBUTING.md#io-contributor-license-agreement).

### CODE REVIEW

**Comment by nicolaiparlog** [comment]
Oh crap, looks like I accidentally activated Travis. 😊 The build fails because upstream changed some interface methods. I think it is ok to merge this PR with build errors, though. 😃

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Contributing
 
+The following guidelines were chosen very deliberately to make sure the project benefits from contributions.
+This is true for such diverse areas as a firm legal foundation or a sensible and helpful commit history.
+
+The guidelines apply to maintainers as well as contributors!
+
 ## Io Contributor License Agreement
```
Comment (smoyer64) [inline CONTRIBUTING.md]: Do we want to scrub the Io name as we go?

Comment (nicolaiparlog) [inline CONTRIBUTING.md]: See #32 for that.

**Comment by nicolaiparlog** [comment]
Awesome, once this is merged, we can actually start making progress. :smile:

**Pull request merged**